improver_tests/wxcode/wxcode/test_wxcode_decision_tree.py

```diff
@@ -131,17 +131,50 @@ def test_node_points_to_valid_value(self):
                 self.assertEqual(fail in tree, True)
 
     def test_diagnostic_len_match(self):
-        """Test diagnostic fields, thresholds and conditions are same len."""
+        """Test diagnostic fields, thresholds and conditions are same
+        structure."""
+
+        def check_value_consistency(values):
+            """Return True if all input lists have same nested list
+            structure. e.g. ['item'] != [['item']]."""
+            type_set = set(map(type, values))
+            if list in type_set:
+                return (
+                        len(type_set) == 1 and
+                        len(set(map(len, values))) == 1 and
+                        all(map(check_value_consistency, zip(*values)))
+                )
+            return True
```

Comment:
This is a non-trivial piece of logic that is duplicated for the global tree. Similarly for other tests in this file. Could we make the tests parametric for decision tree and avoid this duplication?

Reply:
I've refactored this to remove the duplication.

Reply:
thanks, but I meant something more like this: https://github.com/MoseleyS/improver/pull/3/files?diff=unified&w=1